src/s3/requests.c

```diff
@@ -498,11 +503,17 @@ s3_put_object_with_contents(char *objectname, Pointer data, uint64 dataSize,
 	char	   *tmp;
 	int			sc;
 	StringInfoData buf;
-	unsigned char hash[32];
 	long		http_code = 0;
 
-	(void) SHA256((unsigned char *) data, dataSize, hash);
-	contenthash = hex_string((Pointer) hash, sizeof(hash));
+	if (dataHash == NULL)
```

Comment:
maybe if (dataHash == NULL && data != NULL) to not calculating hash of directories

Reply:
For now I'm not sure that checking `data != NULL` is necessary, I left same behavior as it was before:
https://github.com/orioledb/orioledb/blob/3a75cb0bad0804f3aa6748c15f094bf34f4a93bd/src/s3/requests.c#L504
And my experiments show that `SHA256` returns a specific value if `data` is `NULL` and `dataSize` is `0`:
`e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855`.

Reply:
Ok for same behavior as previous.
I also thought of setting SHA256(NULL) as defined constant, and encoded version as a constant too. Maybe apart from this PR.